src/pages/webhooks/xml-schema.md

```diff
@@ -0,0 +1,97 @@
+## hook element
+
+The `hook` element defines the HTTP request to the remote server.
+
+| Attribute | Type   | Description  | Is required | Default |
+|---|---|---|---|---|
+| `name` | String | A hook name that is unique within a batch. | true        | Not applicable     |
+| `url` | String | The HTTP endpoint to send the request for processing. | true        | Not applicable     |
+| `method` | String | The HTTP method, such as POST or PUT, that invokes the hook. | false       | POST    |
+| `priority` | Int    | The priority of the merging hook results in the batch.  | false       | 0       |
+| `required` | Boolean   | Specifies whether hook execution is required or optional. When set to `false` (optional), if the hook fails to execute, the failure is logged and subsequent hooks continue to be processed. When `true`, a failure terminates the process. | false       | true    |
+| `timeout` | Int    | A hard timeout limit (milliseconds) for the request. Requests exceeding this timeout are aborted and logged. | false       | 0       |
+| `softTimeout`| Int    | A soft timeout limit (milliseconds) for the request. Requests exceeding this timeout are logged for debugging purposes. | false       | Not applicable     |
+| `fallbackErrorMessage` | Int    | The error message to return when the hook fails. | false       | Not applicable     |
+| `remove` | Boolean   | Indicates whether to skip a removed hook during the batch execution. | false       | false   |
+| `headers` | Array  | A set of HTTP headers to send with the request. | false       | []      |
+| `fields` | Array  | A set of fields to include in the hook payload. If not set, the entire payload will be sent. | false       | []      |
+
+### headers and header elements
+
+A `headers` element is optional and can contain one or more `header` elements. Each `header` definition creates an HTTP header to be sent in the remote server request.
+
+| Attribute | Type   | Description  | Is required | Default |
+|---|---|---|---|---|
+| `name`     | String  | The header name, in the same form as it will be sent. For example, `Authorization`. Specify the value, such as `Bearer: <token>` in the body of the &lt;header> element. | false       | Not applicable     |
+| `resolver` | String  | The resolver class that appends headers dynamically.         | false       | Not applicable     |
+| `remove`   | Boolean | Set to `true` to remove the header from the request.           | false       | false   |
+
+## fields and field elements
+
+A `fields` element is optional and can contain one or more `field` elements. The `fields` element provides the ability to limit the payload of a webhook to only those fields defined in the individual `field` definitions.
+
+| Attribute | Type   | Description  | Is required | Default |
+|---|---|---|---|---|
+| `name` | String  | The path to the field to include, such as `product.sku`. Do not include the top-level `data` object in the name. | true        | Not applicable     |
```

Comment:
Think my answers on this were confusing. There are cases where `data` should be included in the name. For example, a field like
`<field name='data.product.sku'/>` is valid for an `observer.catalog_product_save_after event `- the unfiltered payload for the event has a top-level `data` key and in this case the name value is being used as the path to the value in the unfiltered payload and as the desired path to the value in the filtered payload.

For plugin events, many wouldn't have payloads with a top-level data key and therefore if the source attribute isn't added it doesn't make much sense to include the top-level data in the name value, but it's not technically disallowed

I'm not sure of the best terms to use, but also wondering if it would be helpful to add to the description that the name is the path in the processed/filtered payload and the source is the path in the unprocessed/unfiltered webhook data payload

Reply:
I forgot to update this line after it became clear to me that you have to know the shape of the data you're getting in the default webhook and the shape of the data you're sending. I changed the example in the "Configure hooks" topic so that it changes the shape and renames a field. So, I think my confusion when we were discussing this area has ben cleared up.